services/account_service.go

```diff
@@ -97,3 +96,41 @@ func (s *AccountAPIService) AccountBalance(
 		},
 	}, nil
 }
+
+// AccountCoins implements /account/coins.
+func (s *AccountAPIService) AccountCoins(
+	ctx context.Context,
+	request *types.AccountCoinsRequest,
+) (*types.AccountCoinsResponse, *types.Error) {
+	if s.config.Mode != configuration.Online {
+		return nil, wrapErr(ErrUnavailableOffline, nil)
+	}
+
+	coins, block, err := s.i.GetCoins(ctx, request.AccountIdentifier)
+	if err != nil {
+		return nil, wrapErr(ErrUnableToGetCoins, err)
+	}
+
+	result := &types.AccountCoinsResponse{
+		BlockIdentifier: block,
+		Coins:           coins,
+	}
+	//@Todo include_mempool query
+
+	if len(request.Currencies) > 0 {
+		filtered := []*types.Coin{}
+		for _, curr := range request.Currencies {
```

Comment:
nit: it may be more performant to create a set `map[string]struct{}{}` where the key is `types.Hash(request.Currency)` and iterate through coins instead of iterating through currencies.

Reply:
Example of using a set: https://github.com/coinbase/rosetta-sdk-go/blob/e6be00ef9da16599780777ca9e3140035f5c6b4b/asserter/account.go#L25-L37